### ISSUE TITLE
[Feature] Configure web servers per project

### ISSUE THREAD

**Issue by lughino** [microsoft/playwright#22496]
I would love to have the possibility to configure local web servers per project.
In my use case, I have a full suite that runs against a local build of my application and a mock server, and a subset of those tests are used as smoke tests against production.

The problem is that playwright always executes the webservers configured when I run only the `smoke-tests` project.

If I had the possibility to configure the `webServer`s per project, I would be able to avoid waiting for the local applications to be instantiated, like:
```ts
export default defineConfig({
  projects: [
    {
      name: 'functional-chromium',
      grepInvert: /@smoke/,
      use: { ...devices['Desktop Chrome'], baseURL: 'http://localhost:3000' },
      webServer: [
        {
           command: 'yarn serve',
           url: 'http://localhost:3000',
           timeout: 380 * 1000,
           reuseExistingServer: true,
        },
        {
           command: 'yarn mock-server',
           port: 3001,
           reuseExistingServer: true,
        },
      ],
    },
    {
       name: 'smoke-chromium',
       grep: /@smoke/,
       use: { ...devices['Desktop Chrome'], baseURL: 'https://production.app.com' },
    },
  ],
});
```

**Comment by aoj2** [comment]
This would indeed be a great feature to have! We are having a lot of different unrelated test projects in one pw-config, to be able to set up different web servers for those would be awesome!

**Comment by mastrzyz** [comment]
Yeah we have a need to have a "Live" project and a project with a Local server. one for CI one for Local so having a way to configure based on project would be great!

**Comment by Diokuz** [comment]
We need this too. We have multiple apps in one repo, and we want to run different servers for different tests.

We dont want to use multiple playwright configs (harder to contribute), or run apps from fixtures.

**Comment by mpreyes** [comment]
Hi, 

any updates here? i'd love this feature too!

**Comment by ashblox** [comment]
+1 on this.

We require the following:
 - E2E configurable to run against local or dev
 - Mocked playwright tests that always target a local web server

I can get part of the way there with a BASE_URL environment variable but I don't want to have to ensure that is always pointing to local when running the mocked tests - those should automatically launch the webserver and point to local.

Right now, my options are: 
1. Split out two config files. One for e2e and one for mocked. Annoying because if I target a specific test on the command line, I have to specify the config file (hard to remember when you're used to just running `npx playwright test path/to/file`
2. Forgo Playwright's built-in management of the web server by handing the spinning up & tearing down of the web server myself, through use of an npm package like concurrently or start-server-and-test.

Curious what other folks workaround are. Would love if people could share.

**Comment by richardkmichael** [comment]
I would like this too.

Specifically, the `webServer.command` property is all I want to change per-project.  I'm testing a server that accepts command line options which affect the generated page under test.  It would be convenient if I could represent different combinations of options as projects as opposed to separate configuration files.

e.g.,
`"command": "npm run server --foo -e BAR=baz"`
`"command": "npm run server --config <FILE>"`

The other `webServer` options are the same.  Although, I'd be fine with repeating the entire webServer object, because of course it can be extracted and used in many places.

**Comment by richarda23** [comment]
+1. for this feature. We have some tests running on local instance of frontend,  but also smoke tests we want to run on a pre-prod deployed app,  for which we don't need local server running at all.
 These smoke tests run at a different stage of the build pipeline, yet we have to build and run the local server even though it's not used

**Comment by kholland950** [comment]
+1. We have two different builds of the same web app, and need to start the web server with different environment variables to test both.

**Comment by jfgreffier** [comment]
Would be useful for component testing as well.  
e.g. In CI I prefer to test a distant website, but use a local server for components
